pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,133 @@
+//       Find a better place for it.
+const useDiscoveryService = (defaultDataProviderUrl) => {
+  const [dataProviderUrl, setDataProviderUrl] = useState(defaultDataProviderUrl)
+  const [discoveredUrls, setDiscoveredUrls] = useState(null)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleDataProviderUrlChange = (newUrl) => {
+    setDiscoveredUrls(null)
+    setDataProviderUrl(newUrl)
+  }
+
+  useDebouncedEffect(() => {
+    if (dataProviderUrl === '') return
+
+    setIsLoading(true)
+
+    // TODO: Call API Here
+    clearTimeout(timeout)
+    timeout = setTimeout(() => {
+      if (dataProviderUrl === 'success.url')
+        setDiscoveredUrls([{ url: dataProviderUrl, isNew: true }])
+      else if (dataProviderUrl === 'duplicate.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }])
+      else if (dataProviderUrl === 'multiple-duplicates.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }, { url: dataProviderUrl }])
+      else setDiscoveredUrls([])
+
+      setIsLoading(false)
+    }, 1000)
+  }, [dataProviderUrl])
+
+  useEffect(() => {
+    setDataProviderUrl(defaultDataProviderUrl)
+  }, [defaultDataProviderUrl])
+
+  return [
+    dataProviderUrl,
+    discoveredUrls,
+    isLoading,
+    handleDataProviderUrlChange,
+  ]
+}
+
+const getHelperMessage = (dataProviderUrl, discoveredUrls) => {
+  if (dataProviderUrl === '' || discoveredUrls === null) {
+    return {
+      message:
+        'It can be any URL: homepage address, any data resource address,' +
+        ' OAI-PMH endpoint or RIOXX endpoint',
+      variant: 'normal',
+    }
+  }
+
+  if (discoveredUrls.length === 0) {
+    return {
+      message:
+        'We could not find any OAI-PMH endpoint.' +
+        ' Please try it with another URL.',
+      variant: 'error',
+    }
+  }
+
+  if (discoveredUrls.length === 1 && discoveredUrls[0].isNew) {
+    return {
+      message: `We found OAI-PMH endpoint at ${discoveredUrls[0].url}`,
+      variant: 'success',
+    }
+  }
+
+  if (discoveredUrls.length === 1) {
+    return {
+      message: 'The repository is already added.',
+      variant: 'success',
+    }
+  }
+
+  return {
+    message:
+      `${dataProviderUrl} and ${discoveredUrls.length} more are already` +
+      ' in our system. If you host more than one repository on the same domain,' +
+      ' please specify exact OAI-PMH endpoint',
+    variant: 'error',
+  }
+}
+
+const DataProviderPage = () => {
+  const router = useRouter()
+  const [
+    dataProviderUrl,
+    discoveredUrls,
+    isLoading,
+    handleDataProviderUrlChange,
+  ] = useDiscoveryService(router.query?.dataProviderUrl || '')
+
+  const handleSubmitForm = () => {
+    // TODO: Handle next step. Show success message
+    // router.push('/next-step')
+  }
+
+  const { message, variant } = getHelperMessage(dataProviderUrl, discoveredUrls)
+
+  return (
+    <DataProviderPageTemplate
+      url={dataProviderUrl}
+      // TODO: Discuss URL validation.
```

Comment:
@viktor-yakubiv What do you think about this comment.

Reply:
I tried to provide my ideas in the review below. Please, resolve if this comment is out of date already.

Basically, we should do URL validation but it can be done on the browser side (type=url) or on the component side using some regexp. I would do it on the browser side since it's not maintained by us and should be more reliable. Also, it's possible to provide a regular expression to the input too. On the other side, the API connecting will be tricky, we will need to manage this state separately (but this maybe is even better, more transparent).